src/Simplify.elm

```diff
@@ -7138,19 +7157,102 @@ listDropChecks : CheckInfo -> Maybe (Error {})
 listDropChecks =
     firstThatConstructsJust
         [ \checkInfo ->
-            case Evaluate.getInt checkInfo checkInfo.firstArg of
-                Just 0 ->
-                    Just
-                        (alwaysReturnsLastArgError
-                            (qualifiedToString checkInfo.fn ++ " 0")
-                            { represents = "list" }
-                            checkInfo
-                        )
+            Evaluate.getInt checkInfo checkInfo.firstArg
+                |> Maybe.andThen
+                    (\count ->
+                        firstThatConstructsJust
+                            [ \() ->
+                                callWithNonPositiveIntCheckErrorSituation
+                                    { int = count, intDescription = "count", fn = checkInfo.fn }
+                                    |> Maybe.map
+                                        (\situation -> alwaysReturnsLastArgError situation listCollection checkInfo)
+                            , \() -> dropOnSmallerCollectionCheck { dropCount = count } listCollection checkInfo
+                            , \() ->
+                                dropOnLargerConstructionFromListLiteralWillRemoveTheseElementsCheck { dropCount = count }
+                                    listCollection
+                                    checkInfo
+                            ]
+                            ()
+                    )
+        , unnecessaryCallOnEmptyCheck listCollection
+        ]
+
+
+{-| The drop check
+
+    drop n (collection with size <= n) --> empty
+
+So for example
+
+    Array.drop 3 (Array.repeat 2 a) --> []
```

Comment:
I think Array does not have a drop-function, only slice.

Reply:
Thanks!